solr/solrj/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -968,7 +968,10 @@ public Replica getLeaderRetry(String collection, String shard, int timeout) thro
         }
         return false;
       });
-    } catch (TimeoutException | InterruptedException e) {
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw e;
```

Comment:
I guess if we are going to throw the `InterruptedException` anyway then why catch it?
The options in my mind are either not catch `InterruptedException` at all, or catch it, mark the thread as interrupted and then throw SolrException

Reply:
And BTW, I'm +1 with letting it bubble up

Reply:
Thanks @tflobbe for the review - I agree and updated